src/NServiceBus.TransportTests/When_failure_happens_after_isolated_send.cs

```diff
@@ -0,0 +1,60 @@
+﻿namespace NServiceBus.TransportTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using NUnit.Framework;
+    using Transport;
+
+    public class When_failure_happens_after_isolated_send : NServiceBusTransportTest
+    {
+        [TestCase(TransportTransactionMode.None)]
+        [TestCase(TransportTransactionMode.ReceiveOnly)]
+        [TestCase(TransportTransactionMode.SendsAtomicWithReceive)]
+        [TestCase(TransportTransactionMode.TransactionScope)]
+        public async Task Should_emit_messages(TransportTransactionMode transactionMode)
+        {
+            var onMessageCalled = new TaskCompletionSource<bool>();
+
+            OnTestTimeout(() => onMessageCalled.SetCanceled());
+
+            await StartPump(async context =>
+                {
+                    if (context.Headers.ContainsKey("CompleteTest"))
+                    {
+                        onMessageCalled.SetResult(true);
+                        return;
+                    }
+
+                    if (context.Headers.ContainsKey("IsolatedSend"))
+                    {
+                        onMessageCalled.SetResult(false);
+                        return;
+                    }
+
+                    await SendMessage(InputQueueName, new Dictionary<string, string>
+                    {
+                        {"IsolatedSend", "true"}
```

Comment:
Yes. The only thing being checked here is the existence of the header. The value associated with the headers could be "asdf" and the test will still pass.